libs/tasty-cannon/src/Test/Tasty/Cannon.hs

```diff
@@ -282,14 +294,25 @@ run (($ Http.defaultRequest) -> ca) uid cid app = liftIO $ do
     stat <- poll wsapp
     case stat of
         Just (Left ex) -> throwIO ex
-        _              -> return wsapp
+        _              -> waitForRegistry numRetries >> return wsapp
   where
     caHost = C.unpack (Http.host ca)
     caPort = Http.port ca
     caPath = "/await" ++ C.unpack (Http.queryString ca)
     caOpts = WS.defaultConnectionOptions
     caHdrs = [ ("Z-User", toByteString' uid), ("Z-Connection", toByteString' cid) ]
 
+    numRetries = 30
+
+    waitForRegistry 0          = throwIO $ RegistrationTimeout numRetries
```

Comment:
i don't fully understand what goes wrong without this delay.  who does what in order to race things against the wall?  perhaps add a source comment here?

Reply:
now that i understand it i don't know what could possibly need documentation any more.  (-:

I have this, hope you like it:

```haskell
-- | Start a client thread in 'Async' that opens a web socket to a Cannon, wait for the connection
-- to register with the Cannon's Gundeck, and return the 'Async' thread.
run :: MonadIO m => Cannon -> UserId -> ConnId -> WS.ClientApp () -> m (Async ())
```

Reply:
👌